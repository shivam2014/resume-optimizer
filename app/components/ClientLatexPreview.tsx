'use client';

import 'katex/dist/katex.min.css';
import Latex from 'react-latex-next';
import { FC, useEffect } from 'react';

interface ClientLatexPreviewProps {
  content: string;
  className?: string;
  templateStyles?: string;
}

const ClientLatexPreview: FC<ClientLatexPreviewProps> = ({ 
  content, 
  className = '',
  templateStyles 
}) => {
  useEffect(() => {
    // Create a style element for the template styles
    if (templateStyles) {
      const styleElement = document.createElement('style');
      styleElement.textContent = templateStyles;
      document.head.appendChild(styleElement);

      return () => {
        document.head.removeChild(styleElement);
      };
    }
  }, [templateStyles]);

  return (
    <div className={`latex-preview ${className}`}>
      <Latex>{content}</Latex>
    </div>
  );
};

export default ClientLatexPreview;