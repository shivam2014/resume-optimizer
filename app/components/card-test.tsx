import { Card, CardContent } from "@/components/ui/card";

export default function CardTest() {
  return (
    <Card className="w-[350px]">
      <CardContent className="p-4">
        <h2 className="text-xl font-semibold">Test Card</h2>
        <p className="text-sm text-muted-foreground">
          This is a test of the card component
        </p>
      </CardContent>
    </Card>
  );
}