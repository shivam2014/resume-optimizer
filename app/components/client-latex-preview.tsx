import { useEffect, useState } from 'react';
import 'katex/dist/katex.min.css';
import Latex from 'react-latex-next';
import { cn } from '@/lib/utils';

interface ClientLatexPreviewProps {
  content: string;
  className?: string;
  templateStyles?: string;
}

const ClientLatexPreview: React.FC<ClientLatexPreviewProps> = ({
  content,
  className,
  templateStyles
}) => {
  const [isClient, setIsClient] = useState(false);
  const [error, setError] = useState<string | null>(null);

  useEffect(() => {
    setIsClient(true);
  }, []);

  if (!isClient) {
    return (
      <div className="w-full h-full animate-pulse bg-gray-100 rounded-lg">
        <div className="h-full flex items-center justify-center">
          <div className="w-6 h-6 border-2 border-primary border-t-transparent rounded-full animate-spin" />
        </div>
      </div>
    );
  }

  try {
    return (
      <div className={cn("latex-preview", className)}>
        {templateStyles && (
          <style dangerouslySetInnerHTML={{ __html: templateStyles }} />
        )}
        <Latex>{content}</Latex>
      </div>
    );
  } catch (err) {
    console.error('LaTeX rendering error:', err);
    return (
      <div className="text-red-600 p-4 text-center">
        Failed to render LaTeX content. Please check the syntax.
      </div>
    );
  }
};

export default ClientLatexPreview;