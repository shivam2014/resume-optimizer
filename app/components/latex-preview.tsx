import { useEffect, useState } from 'react';
import { parse, HtmlGenerator } from 'latex.js';
import { Skeleton } from '@/components/ui/skeleton';

interface LatexPreviewProps {
  template: string;
  content: string;
}

const LatexPreview: React.FC<LatexPreviewProps> = ({ template, content }) => {
  const [previewHtml, setPreviewHtml] = useState<string>('');
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [error, setError] = useState<string | null>(null);

  useEffect(() => {
    const generatePreview = async () => {
      try {
        setIsLoading(true);
        setError(null);

        // Load template content
        const response = await fetch(`/api/latex/template?path=${encodeURIComponent(template)}`);
        if (!response.ok) {
          throw new Error('Failed to load template');
        }
        const templateContent = await response.text();
        
        // Replace content placeholder with actual content
        const latexContent = templateContent.replace('%RESUME_CONTENT%', content);
        
        // Create LaTeX parser with custom configuration
        const generator = new HtmlGenerator({
          hyphenate: false,
          styleSheet: [
            '/latex-styles.css',
          ],
        });
        const parser = parse(latexContent, { generator });
        
        // Generate HTML preview with proper styling
        const html = `
          ${generator.stylesAndScripts()}
          <div class="latex-preview-container">
            ${generator.documentFragment.innerHTML}
          </div>
        `;
        setPreviewHtml(html);
      } catch (err) {
        console.error('LaTeX preview generation failed:', err);
        setError('Failed to generate preview. Please check your LaTeX syntax.');
      } finally {
        setIsLoading(false);
      }
    };

    if (template && content) {
      const timeoutId = setTimeout(generatePreview, 500);
      return () => clearTimeout(timeoutId);
    }
  }, [template, content]);

  if (isLoading) {
    return (
      <div className="w-full h-[500px] bg-white rounded-lg p-4">
        <Skeleton className="w-full h-full" />
      </div>
    );
  }

  if (error) {
    return (
      <div className="w-full h-[500px] bg-red-50 p-4 rounded-lg border border-red-200">
        <p className="text-red-600">{error}</p>
      </div>
    );
  }

  return (
    <div 
      className="w-full h-[500px] overflow-auto bg-white p-4 rounded-lg shadow-sm"
      dangerouslySetInnerHTML={{ __html: previewHtml }}
    />
  );
};

export default LatexPreview;