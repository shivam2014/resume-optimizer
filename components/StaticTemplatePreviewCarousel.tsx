import { useState } from 'react';
import Image from 'next/image';
import { Button } from '@/components/ui/button';
import { Card } from '@/components/ui/card';
import { ChevronLeft, ChevronRight } from 'lucide-react';
import { cn } from '@/lib/utils';
import { TemplateMetadata } from '@/types/templates';

interface StaticTemplatePreviewCarouselProps {
  templates: TemplateMetadata[];
}

export function StaticTemplatePreviewCarousel({ templates }: StaticTemplatePreviewCarouselProps) {
  const [currentIndex, setCurrentIndex] = useState(0);
  const [isLoading, setIsLoading] = useState(false);
  const [hasError, setHasError] = useState(false);

  if (!templates.length) {
    return (
      <Card className="w-full h-64 flex items-center justify-center">
        <p className="text-muted-foreground">No template selected</p>
      </Card>
    );
  }

  const currentTemplate = templates[currentIndex];

  const handlePrevious = () => {
    setIsLoading(true);
    setCurrentIndex((prev) => (prev === 0 ? templates.length - 1 : prev - 1));
  };

  const handleNext = () => {
    setIsLoading(true);
    setCurrentIndex((prev) => (prev === templates.length - 1 ? 0 : prev + 1));
  };

  const handleImageLoad = () => {
    setIsLoading(false);
    setHasError(false);
  };

  const handleImageError = () => {
    setIsLoading(false);
    setHasError(true);
  };

  const imagePath = currentTemplate.previewImage
    ? `/templates/latex/${currentTemplate.previewImage}`
    : `/templates/latex/${currentTemplate.name.toLowerCase()}.png`;

  return (
    <div className="relative w-full">
      <Card className="w-full aspect-[16/9] relative overflow-hidden">
        {isLoading && (
          <div role="status" className="absolute inset-0 bg-background/80 flex items-center justify-center">
            <div className="animate-spin h-8 w-8 border-4 border-primary border-t-transparent rounded-full" />
          </div>
        )}
        
        <Image
          src={imagePath}
          alt={`Preview of ${currentTemplate.name} template`}
          fill
          className={cn(
            "object-cover transition-opacity duration-300",
            isLoading ? "opacity-0" : "opacity-100"
          )}
          onLoadingComplete={handleImageLoad}
          onError={handleImageError}
        />
        
        {hasError && (
          <div className="absolute inset-0 flex items-center justify-center bg-muted">
            <p className="text-muted-foreground">Preview image not available</p>
          </div>
        )}
      </Card>

      <div className="absolute inset-y-0 left-0 flex items-center">
        <Button
          variant="ghost"
          size="icon"
          onClick={handlePrevious}
          className="h-8 w-8 rounded-full bg-background/80 hover:bg-background/90"
          aria-label="Previous template"
        >
          <ChevronLeft className="h-4 w-4" />
        </Button>
      </div>

      <div className="absolute inset-y-0 right-0 flex items-center">
        <Button
          variant="ghost"
          size="icon"
          onClick={handleNext}
          className="h-8 w-8 rounded-full bg-background/80 hover:bg-background/90"
          aria-label="Next template"
        >
          <ChevronRight className="h-4 w-4" />
        </Button>
      </div>

      <div className="absolute bottom-4 left-0 right-0 text-center">
        <p className="text-sm font-medium text-white bg-background/80 inline-block px-2 py-1 rounded">
          {currentTemplate.name}
        </p>
      </div>
    </div>
  );
}