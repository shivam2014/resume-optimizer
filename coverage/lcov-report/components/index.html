
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> components</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">47.77% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>193/404</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">35.46% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>61/172</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">36.61% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>26/71</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">47.61% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>190/399</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="StaticTemplatePreviewCarousel.tsx"><a href="StaticTemplatePreviewCarousel.tsx.html">StaticTemplatePreviewCarousel.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	<td data-value="81.81" class="pct high">81.81%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	</tr>

<tr>
	<td class="file high" data-value="comparison-view.tsx"><a href="comparison-view.tsx.html">comparison-view.tsx</a></td>
	<td data-value="80" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 80%"></div><div class="cover-empty" style="width: 20%"></div></div>
	</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="20" class="abs high">16/20</td>
	<td data-value="42.85" class="pct low">42.85%</td>
	<td data-value="14" class="abs low">6/14</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="20" class="abs high">16/20</td>
	</tr>

<tr>
	<td class="file medium" data-value="error-boundary.tsx"><a href="error-boundary.tsx.html">error-boundary.tsx</a></td>
	<td data-value="72.72" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 72%"></div><div class="cover-empty" style="width: 28%"></div></div>
	</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	</tr>

<tr>
	<td class="file low" data-value="file-uploader.tsx"><a href="file-uploader.tsx.html">file-uploader.tsx</a></td>
	<td data-value="35.55" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 35%"></div><div class="cover-empty" style="width: 65%"></div></div>
	</td>
	<td data-value="35.55" class="pct low">35.55%</td>
	<td data-value="45" class="abs low">16/45</td>
	<td data-value="14.28" class="pct low">14.28%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="35.55" class="pct low">35.55%</td>
	<td data-value="45" class="abs low">16/45</td>
	</tr>

<tr>
	<td class="file low" data-value="latex-preview.tsx"><a href="latex-preview.tsx.html">latex-preview.tsx</a></td>
	<td data-value="28.57" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 28%"></div><div class="cover-empty" style="width: 72%"></div></div>
	</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	</tr>

<tr>
	<td class="file low" data-value="resume-editor.tsx"><a href="resume-editor.tsx.html">resume-editor.tsx</a></td>
	<td data-value="27.77" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 27%"></div><div class="cover-empty" style="width: 73%"></div></div>
	</td>
	<td data-value="27.77" class="pct low">27.77%</td>
	<td data-value="36" class="abs low">10/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="35" class="abs low">10/35</td>
	</tr>

<tr>
	<td class="file low" data-value="resume-optimizer.tsx"><a href="resume-optimizer.tsx.html">resume-optimizer.tsx</a></td>
	<td data-value="44.62" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 44%"></div><div class="cover-empty" style="width: 56%"></div></div>
	</td>
	<td data-value="44.62" class="pct low">44.62%</td>
	<td data-value="251" class="abs low">112/251</td>
	<td data-value="39.63" class="pct low">39.63%</td>
	<td data-value="111" class="abs low">44/111</td>
	<td data-value="41.17" class="pct low">41.17%</td>
	<td data-value="34" class="abs low">14/34</td>
	<td data-value="44.57" class="pct low">44.57%</td>
	<td data-value="249" class="abs low">111/249</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2025-03-21T10:32:49.183Z
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    