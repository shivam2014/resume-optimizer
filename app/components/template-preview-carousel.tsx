import { useState, useEffect } from 'react';
import { Carousel, CarouselContent, CarouselItem, CarouselPrevious, CarouselNext } from '@/components/ui/carousel';
import { Card, CardContent } from '@/components/ui/card';
import { TemplateMetadata, getTemplates, getSampleContent } from '@/lib/template-config';
import { cn } from '@/lib/utils';
import ClientLatexPreview from './client-latex-preview';

interface TemplatePreviewCarouselProps {
  templates?: TemplateMetadata[];
  content?: string;
  onTemplateSelect?: (template: TemplateMetadata) => void;
}

const TemplatePreviewCarousel: React.FC<TemplatePreviewCarouselProps> = ({
  templates = getTemplates(),
  content = getSampleContent(),
  onTemplateSelect
}) => {
  const [activeTemplate, setActiveTemplate] = useState<string>(templates[0].path);
  const [templateStyles, setTemplateStyles] = useState<Map<string, string>>(new Map());
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const loadTemplateStyles = async () => {
      const styles = new Map<string, string>();
      
      for (const template of templates) {
        try {
          const response = await fetch(`/templates/latex/${template.path}/style.css`);
          if (response.ok) {
            const css = await response.text();
            styles.set(template.path, css);
          }
        } catch (err) {
          console.warn(`Failed to load styles for template ${template.path}:`, err);
        }
      }

      setTemplateStyles(styles);
      setLoading(false);
    };

    loadTemplateStyles();
  }, [templates]);

  const handleTemplateSelect = (template: TemplateMetadata) => {
    setActiveTemplate(template.path);
    onTemplateSelect?.(template);
  };

  return (
    <div className="w-full space-y-6">
      <Carousel className="w-full group">
        <CarouselContent className="-ml-2 md:-ml-4">
          {templates.map((template) => (
            <CarouselItem
              key={template.path}
              className="pl-2 md:pl-4 basis-full md:basis-1/2 lg:basis-1/3"
            >
              <div className="relative aspect-[210/297] group">
                <Card
                  className={cn(
                    "h-full cursor-pointer transition-all duration-300 hover:shadow-lg overflow-hidden",
                    activeTemplate === template.path && "ring-2 ring-primary"
                  )}
                  onClick={() => handleTemplateSelect(template)}
                  aria-label={`Preview of ${template.name} template`}
                >
                  <CardContent className="p-4 h-full">
                    <ClientLatexPreview
                      content={content}
                      templateStyles={templateStyles.get(template.path)}
                      className="transform scale-[0.4] origin-top-left absolute inset-0 w-[250%] h-[250%]"
                    />
                    <div className="absolute inset-0 bg-black/60 opacity-0 group-hover:opacity-100 transition-opacity rounded-lg flex flex-col items-center justify-center p-4 text-white">
                      <h3 className="text-lg font-semibold text-center mb-2">
                        {template.name}
                      </h3>
                      {template.description && (
                        <p className="text-sm text-center text-gray-200">
                          {template.description}
                        </p>
                      )}
                      {template.source && (
                        <a
                          href={template.source}
                          target="_blank"
                          rel="noopener noreferrer"
                          className="mt-4 text-sm text-blue-300 hover:text-blue-200 underline"
                          onClick={(e) => e.stopPropagation()}
                        >
                          View Source
                        </a>
                      )}
                    </div>
                  </CardContent>
                </Card>
              </div>
            </CarouselItem>
          ))}
        </CarouselContent>
        <div className="hidden md:block">
          <CarouselPrevious className="bg-white hover:bg-gray-50" />
          <CarouselNext className="bg-white hover:bg-gray-50" />
        </div>
      </Carousel>

      <div className="flex justify-center gap-2">
        {templates.map((template) => (
          <button
            key={template.path}
            onClick={() => handleTemplateSelect(template)}
            className={cn(
              "w-2 h-2 rounded-full transition-all duration-300",
              activeTemplate === template.path
                ? "bg-primary w-6"
                : "bg-gray-300 hover:bg-gray-400"
            )}
            aria-label={`Select ${template.name}`}
          />
        ))}
      </div>
    </div>
  );
};

export default TemplatePreviewCarousel;