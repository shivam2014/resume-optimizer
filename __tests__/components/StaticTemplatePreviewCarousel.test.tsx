import { render, screen, fireEvent, waitFor, cleanup } from '@testing-library/react';
import { StaticTemplatePreviewCarousel } from '@/components/StaticTemplatePreviewCarousel';

// Mock templates with realistic data
const mockTemplates = [
  { id: 'modern-1', name: 'Modern Template' },
  { id: 'professional-2', name: 'Professional Template' },
  { id: 'creative-3', name: 'Creative Template' },
];

// Mock next/image
jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line @next/next/no-img-element
    return <img
      data-testid="next-image"
      src={props.src}
      alt={props.alt}
      onLoad={props.onLoadingComplete}
      onError={props.onError}
    />
  },
}));

describe('StaticTemplatePreviewCarousel', () => {
  // Cleanup after each test
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it('renders correctly with template data', () => {
    render(<StaticTemplatePreviewCarousel templates={mockTemplates} />);
    
    expect(screen.getByText('Modern Template')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /previous template/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /next template/i })).toBeInTheDocument();
    expect(screen.getByTestId('next-image')).toHaveAttribute('src', '/templates/modern-1.png');
  });

  it('handles empty templates array', () => {
    render(<StaticTemplatePreviewCarousel templates={[]} />);
    
    expect(screen.queryByRole('button')).not.toBeInTheDocument();
    expect(screen.getByText(/no template selected/i)).toBeInTheDocument();
  });

  describe('Navigation', () => {
    it('cycles through templates forward', async () => {
      render(<StaticTemplatePreviewCarousel templates={mockTemplates} />);
      
      const nextButton = screen.getByRole('button', { name: /next template/i });
      
      // First template
      expect(screen.getByText('Modern Template')).toBeInTheDocument();
      
      // Go to second template
      fireEvent.click(nextButton);
      await waitFor(() => {
        expect(screen.getByText('Professional Template')).toBeInTheDocument();
      });
      
      // Go to third template
      fireEvent.click(nextButton);
      await waitFor(() => {
        expect(screen.getByText('Creative Template')).toBeInTheDocument();
      });
      
      // Cycle back to first template
      fireEvent.click(nextButton);
      await waitFor(() => {
        expect(screen.getByText('Modern Template')).toBeInTheDocument();
      });
    });

    it('cycles through templates backward', async () => {
      render(<StaticTemplatePreviewCarousel templates={mockTemplates} />);
      
      const prevButton = screen.getByRole('button', { name: /previous template/i });
      
      // First template
      expect(screen.getByText('Modern Template')).toBeInTheDocument();
      
      // Go to last template
      fireEvent.click(prevButton);
      await waitFor(() => {
        expect(screen.getByText('Creative Template')).toBeInTheDocument();
      });
    });
  });

  describe('Loading states', () => {
    it('shows loading state when changing templates', async () => {
      render(<StaticTemplatePreviewCarousel templates={mockTemplates} />);
      
      const nextButton = screen.getByRole('button', { name: /next template/i });
      fireEvent.click(nextButton);
      
      // Check loading state appears
      expect(screen.getByRole('status')).toBeInTheDocument();
      
      // Simulate image load
      const image = screen.getByTestId('next-image');
      fireEvent.load(image);
      
      // Check loading state disappears
      await waitFor(() => {
        expect(screen.queryByRole('status')).not.toBeInTheDocument();
      });
    });

    it('handles image load success correctly', async () => {
      render(<StaticTemplatePreviewCarousel templates={mockTemplates} />);
      
      const image = screen.getByTestId('next-image');
      fireEvent.load(image);
      
      expect(screen.queryByRole('status')).not.toBeInTheDocument();
      expect(screen.queryByText(/failed to load/i)).not.toBeInTheDocument();
    });

    it('handles image load error correctly', async () => {
      render(<StaticTemplatePreviewCarousel templates={mockTemplates} />);
      
      const image = screen.getByTestId('next-image');
      fireEvent.error(image);
      
      expect(screen.queryByRole('status')).not.toBeInTheDocument();
      expect(screen.getByText(/failed to load template preview/i)).toBeInTheDocument();
    });
  });

  describe('Edge cases', () => {
    it('handles single template correctly', () => {
      const singleTemplate = [mockTemplates[0]];
      render(<StaticTemplatePreviewCarousel templates={singleTemplate} />);
      
      // Navigation should still work (cycle through single template)
      const nextButton = screen.getByRole('button', { name: /next template/i });
      fireEvent.click(nextButton);
      
      expect(screen.getByText('Modern Template')).toBeInTheDocument();
    });

    it('handles rapid navigation clicks', async () => {
      render(<StaticTemplatePreviewCarousel templates={mockTemplates} />);
      
      const nextButton = screen.getByRole('button', { name: /next template/i });
      
      // Click multiple times rapidly
      fireEvent.click(nextButton);
      fireEvent.click(nextButton);
      fireEvent.click(nextButton);
      
      // Should end up at the correct template
      await waitFor(() => {
        expect(screen.getByText('Modern Template')).toBeInTheDocument();
      });
    });
  });

  it('matches snapshot', () => {
    const { container } = render(
      <StaticTemplatePreviewCarousel templates={mockTemplates} />
    );
    expect(container).toMatchSnapshot();
  });
});