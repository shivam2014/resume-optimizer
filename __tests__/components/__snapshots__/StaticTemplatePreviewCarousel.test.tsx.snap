// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticTemplatePreviewCarousel matches snapshot 1`] = `
<div>
  <div
    class="relative w-full"
  >
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm w-full aspect-[16/9] relative overflow-hidden"
    >
      <img
        alt="Modern Template"
        data-testid="next-image"
        src="/templates/modern-1.png"
      />
    </div>
    <div
      class="absolute inset-y-0 left-0 flex items-center"
    >
      <button
        aria-label="Previous template"
        class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 hover:text-accent-foreground h-8 w-8 rounded-full bg-background/80 hover:bg-background/90"
      >
        <svg
          class="lucide lucide-chevron-left h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m15 18-6-6 6-6"
          />
        </svg>
      </button>
    </div>
    <div
      class="absolute inset-y-0 right-0 flex items-center"
    >
      <button
        aria-label="Next template"
        class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 hover:text-accent-foreground h-8 w-8 rounded-full bg-background/80 hover:bg-background/90"
      >
        <svg
          class="lucide lucide-chevron-right h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m9 18 6-6-6-6"
          />
        </svg>
      </button>
    </div>
    <div
      class="absolute bottom-4 left-0 right-0 text-center"
    >
      <p
        class="text-sm font-medium text-white bg-background/80 inline-block px-2 py-1 rounded"
      >
        Modern Template
      </p>
    </div>
  </div>
</div>
`;
